### DISCUSSION TITLE
RapidOrientation表格旋转角度识别错误

### DISCUSSION THREAD

**Discussion by Dkaaaaa** [RapidAI/RapidOrientation#32] [Q&A]
您好，我使用RapidOrientation对以下表格进行角度识别，但是结果显示为180度，实际应该是270度，想知道是什么原因引起的？我尝试过缩小图片分辨率到1500左右，然后结果就变成了90度。。。
![imgf000243_0001](https://github.com/user-attachments/assets/23e11e59-e995-4923-a970-52f593aee25f)

**Comment by Joker1212** [comment]
首先90度是正确的，得到的角度代表的我记得是你后续处理应该旋转的角度，你可以多试试几张，RapidOrientation原理是裁剪图片部分进行进行大角度预测，为了稳定性进行了多次预测，具体可能得看源码了，因为不是我这边开发的，建议是可以多次执行取置信度+次数最高的角度

**Reply by Dkaaaaa** [reply]
您好，270度才是正确的，我测试了其它跟这个表格方向一致的图片，结果为270度，然后用opencv进行旋转，输入的角度也是270度，得到了水平方向的表格

**Reply by Joker1212** [reply]
ok,可能我记错了，找到办法就行，为了稳定性还是建议多执行几次检测，可以看看内部是不是随机裁剪识别的

**Comment by Joker1212** [comment]
另外可以使用RapidTableDetection，进行目标检测和角度分类，并裁剪得到表格区域（因为没有透视和小角度，不建议使用边缘检测）

**Reply by Dkaaaaa** [reply]
您好，我用RapidOrientation进行了多次检测，表格角度的识别结果还是没有变化。于是我使用您建议的RapidTableDetection，以上图片运行两次之后可以得到矫正角度后即水平方向的表格。但是另外有一些图片本来是水平的，但是RapidTableDetection还是会把他旋转180度，并且我多次运行后还是不能矫正为水平方向。想知道还有什么方向可以尝试的吗？谢谢

**Reply by Dkaaaaa** [reply]
或者是否可以先对角度进行识别，如果是水平则不进行后续的矫正呢

**Reply by Joker1212** [reply]
emmm,精度问题的话，要么微调训练，要么就是你说的识别，我想知道这个运行两次是什么意思，RapidTableDetection应该是一次就能提取出来的吧

**Reply by Dkaaaaa** [reply]
- 首先，我的图片一般都是一张表格，我的需求只是对图片中的表格方向进行纠正，于是我只使用了RapidTableDetection的方向分类功能。
- 运行两次就是假设我是图片A去运行RapidTableDetection得到图片B，然后我继续用图片B去运行RapidTableDetection得到图片C。
- 因为您上面提到：为了稳定性还是建议多执行几次检测，可以看看内部是不是随机裁剪识别的，我不知道RapidTableDetection是不是也是有这种随机性，但是我测试发现有一些图片如果第一次没有纠正成功，再运行几次就可以纠正成功了。

**Reply by Joker1212** [reply]
。。。这确实是意外收获哈，RapidTableDetection没有随机截图,每次都是整图

**Reply by Dkaaaaa** [reply]
我把您项目上给的代码简单封装了一下，除了use_obj_det、use_edge_det、use_cls_det其余均为默认参数，即以下的rotateImage2函数，该函数会保存纠正后的图片以及返回纠正后的图片的文件路径
![imgf000169_0001](https://github.com/user-attachments/assets/42f43134-0eab-4501-87d3-3da39bf6b410)
对于以上图片，如果我运行
```
file_path = rotateImage2(file, use_obj_det=False, use_edge_det=False, use_cls_det=True)
file_path = rotateImage2(file_path, use_obj_det=True, use_edge_det=False, use_cls_det=True)
```
那么会得到下面这个：
![imgf000169_0001-rotated](https://github.com/user-attachments/assets/6bd42e38-fb20-4582-bae2-e0438a47dda4)
如果我运行
```
file_path = rotateImage2(file, use_obj_det=True, use_edge_det=False, use_cls_det=True)
file_path = rotateImage2(file_path, use_obj_det=True, use_edge_det=False, use_cls_det=True)
```
或者
```
file_path = rotateImage2(file, use_obj_det=True, use_edge_det=False, use_cls_det=True)
```
则会得到这个
![imgf000169_0001-rotated](https://github.com/user-attachments/assets/11931c23-b757-4e62-8fd0-fc6e89d4219f)
区别只是use_obj_det这个参数，真的好好奇，为什么会这样子

**Reply by Joker1212** [reply]
可以参考readme文档哈，这三个模型针对不同的任务，use_obj_det还是需要的，因为是表格的目标检测
